meson.build

```
@@ -143,9 +143,14 @@ configh_data.set('_CRT_NONSTDC_NO_DEPRECATE', 1)
 configh_data.set('WIN32_LEAN_AND_MEAN', 1)
 
 # Supports -Wl,--version-script?
+if cc.has_link_argument('-Wl,--undefined-version')
+    extra_linker_args = ',--undefined-version'
+else
+    extra_linker_args = ''
+endif
 have_version_script = cc.links(
     'int main(){}',
-    args: '-Wl,--undefined-version,--version-script=' + meson.current_source_dir()/'xkbcommon.map',
+    args: '-Wl,--version-script=' + meson.current_source_dir()/'xkbcommon.map' + extra_linker_args,
```

fwiw, meson has for quite a while supported f-strings so this may be easier as:
```suggestion
    args: f'-Wl,--version-script=@map@@extra_linker_args@',
```
with an extra
```
map = meson.current_source_dir()/'xkbcommon.map
```
above somewhere

---

@whot Good idea. I raised minimal meson version to from `0.52` (October 2019) to `0.58` (May 2021) to enable f-strings.

---

Given that #313 updated to a meson version of barely one year old at that time, I guess 3 years old is OK here.